AsyncDisplayKit/Details/_ASDisplayLayer.mm

```diff
@@ -122,6 +122,12 @@ - (void)setNeedsLayout
   ASDisplayNodeAssertMainThread();
   [super setNeedsLayout];
 }
+
+- (void)layoutIfNeeded
+{
+  ASDisplayNodeAssertMainThread();
+  [super layoutIfNeeded];
```

Comment:
Why not also trampoline this one with perform-on-main, if we do for the View class?